src/syrupy/location.py

```diff
@@ -17,8 +19,12 @@ def __init__(self, node: Any):
 
     @property
     def classname(self) -> Optional[str]:
-        _, __, qualname = self._node.location
-        return ".".join(list(self.__valid_ids(qualname))[:-1]) or None
+        return (
+            ".".join(
+                node.name for node in self._node.listchain() if isinstance(node, Class)
+            )
+            or None
+        )
```

Comment:
Would prefer we fixed the `__valid_ids` method so both `__parse` and `classname` are using the same logic

Reply:
I suspect `classname` doesn't need that logic – unlike with the others, in this context, we know that we have some set of valid Python identifiers anyway. It seems like it might, then, be slightly better to actually look for classes in the node hierarchy, as per this implementation?

Reply:
Hmm missed this yesterday, it's an easy enough refactor to revert. @noahnu what do you think?

Reply:
Right, I mean, that's what this implementation does – `listchain` shows all the current "nodes" in the test chain (modules, classes, &c.), so I'm going off that directly.

The alternative is just to split on `[` and ignoring the part after that in `__valid_ids` before splitting on `.`, which gets you to the same place.